src/main/scala/nl.knaw.dans.easy.solr4files/components/DDM.scala

```diff
@@ -0,0 +1,120 @@
+            ("dataset_abr_subject", n.text), // TODO https://github.com/DANS-KNAW/easy-dtap/pull/131#pullrequestreview-60582037
+            ("dataset_subject", map.getOrElse(n.text, ""))
+          )
+        }
+      } ++
+      (dcmiMetadata \ "relation").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "conformsTo").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "isVersionOf").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "hasVersion").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "isReplacedBy").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "replaces").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "isRequiredBy").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "requires").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "isPartOf").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "hasPart").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "isReferencedBy").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "references").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "isFormatOf").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "hasFormat").map(n => ("dataset_relation", n.text)) ++
+      (dcmiMetadata \ "temporal").map(n => ("dataset_coverage", n.text))
+  // TODO spatial https://github.com/DANS-KNAW/easy-schema/blob/acb6506/src/main/assembly/dist/docs/examples/ddm/example2.xml#L280-L320
+  // TODO   "dataset_coverage" -> (dcmiMetadata \ "spatial").text,
+}
+
+object DDM {
+
+  private val xsiURI = "http://www.w3.org/2001/XMLSchema-instance"
+
+  private val abrPrefix = "abr:ABR"
+
+  private lazy val abrMaps = loadVocabularies("https://easy.dans.knaw.nl/schemas/vocab/2012/10/abr-type.xsd")
+    .map { case (k, v) => // attributes in xsd are complex/periode
+      (s"$abrPrefix$k", v) // we want to search with DDM attributes which are abr:ABRcomplex/abr:ABRperiode
+    }
+
+  private lazy val audienceMap = loadVocabularies(
+    "https://easy.dans.knaw.nl/schemas/vocab/2015/narcis-type.xsd"
+  )("Discipline")
+
+  private def getAbrMap(ddmSubjectNode: Node): Option[Map[String, String]] = {
+    ddmSubjectNode
+      .attribute(xsiURI, "type")
+      .map(_.text)
+      .filter(_.startsWith(abrPrefix))
+      .flatMap(abrMaps.get)
+  }
+
+  private def isDOI(identifierNode: Node) = {
+    identifierNode
+      .attribute(xsiURI, "type")
+      .map(_.text)
+      .contains("id-type:DOI")
+  }
+
+  private def loadVocabularies(xsdURL: String): Map[String, Map[String, String]] = {
+    for {
+      url <- Try(new URL(xsdURL))
+      xml <- url.loadXml
+    } yield (xml \ "simpleType")
+      .map(n => (n.attribute("name").map(_.text).getOrElse(""), findKeyValuePairs(n)))
+      .toMap
+  }.getOrElse(Map.empty)
+
+  private def findKeyValuePairs(table: Node): Map[String, String] = {
+    (table \\ "enumeration")
```

Comment:
You put all codes in one map, so both complex and period types. Now the only overlap seems to be 'XXX', being 'Onbekend', but it might be safer to use two maps and make it simpler to treat them differently later on.

Reply:
the caller takes care of the different tables